memcached/init.lua

```diff
@@ -185,6 +185,7 @@ local typetable = {
 }
 
 local err_enomem          = "Can't allocate memory for %s"
+local err_bad_box         = "Tarantool must be initialized with box.cfg()"
```

Comment:
A terminological nitpicking: tarantool is an application server and a database. `box.cfg()` initializes the dababase part, while the whole application server can work without this.

I would reflect Tarantool's message here:

```lua
tarantool> (select(2, pcall(function() return box.info end)):gsub('^[^ ]+ ([^\n]+).*$', '%1'))
---
- Please call box.cfg{} first
...
```

Reply:
Thanks for suggestion!

I updated a patch:
```lua
--- a/memcached/init.lua
+++ b/memcached/init.lua        
@@ -185,7 +185,6 @@ local typetable = {
 }            
              
 local err_enomem          = "Can't allocate memory for %s"
-local err_bad_box         = "Tarantool must be initialized with box.cfg()"
 local err_bad_args        = "Arguments must be in dictionary"
 local err_no_such_option  = "No such option '%s'"
 local err_bad_option_type = "Bad '%s' option type, expected '%s', got '%s'"
(1/2) Stage this hunk [y,n,q,a,d,j,J,g,/,e,?]? y                                                                                        
@@ -340,8 +339,9 @@ local function memcached_init(name, uri, opts)
         end                    
         return memcached_services[name]  
     end
-    if type(box.cfg) == 'function' then
-        error(fmt(err_bad_box, name))
+    local _, err = pcall(function() return box.info end)
+    if err ~= nil then
+        error(err:gsub('^[^ ]+ ([^\n]+).*$', '%1'))
     end
     local conf = config_initial(opts)
     local instance = {}
```

Please you advice: can I rely on regexp or may be better to show my own error message?

Reply:
changed:

```lua
--- a/memcached/init.lua
+++ b/memcached/init.lua
@@ -186,6 +186,7 @@ local typetable = {
 
 local err_enomem          = "Can't allocate memory for %s"
 local err_bad_args        = "Arguments must be in dictionary"
+local err_bad_box         = "Please call box.cfg{} first"
 local err_no_such_option  = "No such option '%s'"
 local err_bad_option_type = "Bad '%s' option type, expected '%s', got '%s'"
 local err_bad_value       = "Bad value for argument '%s'"
@@ -339,6 +340,9 @@ local function memcached_init(name, uri, opts)
         end
         return memcached_services[name]
     end
+    if type(box.cfg) == 'function' then
+        error(fmt(err_bad_box, name))
+    end
     local conf = config_initial(opts)
     local instance = {}
     instance.opts = conf
```

Reply:
Thanks!